### PR TITLE
executor: Fix the bug in the statement of "on duplicate..." 

### PR DESCRIPTION
**Pull request by zimulala**
Fix the bug in the statement of "on duplicate..." when updating the PK handle.

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @coocood @shenli @tiancaiamao

File: executor/executor_write_test.go [hunk 1]
```diff
@@ -94,8 +94,14 @@ func (s *testSuite) TestInsert(c *C) {
 	c.Assert(err, NotNil)
 	tk.MustExec("rollback")
 
-	insertSQL := `insert into insert_test (id, c2) values (1, 1) on duplicate key update c2=10;`
+	// Updating column is PK handle.
+	r := tk.MustQuery("select * from insert_test where id = 1;")
+	rowStr := fmt.Sprintf("%v %v %v %v", "1", "1", nil, "1")
```
Comment (coocood) [inline executor/executor_write_test.go]: missing a check?

**Comment by shenli** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**